datafusion/expr/src/expr_rewriter.rs

```diff
@@ -519,6 +495,49 @@ pub fn unnormalize_cols(exprs: impl IntoIterator<Item = Expr>) -> Vec<Expr> {
     exprs.into_iter().map(unnormalize_col).collect()
 }
 
+struct RewriterAdapter<F> {
```

Comment:
It is better to add some comments for `RewriterAdapter`, but this is not a blocker.

Reply:
Good idea, added in 34a725c2b